src/client/java/net/neoforged/neoforge/client/neo3d/opengl/NeoGlDeviceFeatures.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.neo3d.opengl;
+
+import net.minecraft.Util;
+import net.neoforged.neoforge.client.neo3d.GpuDeviceFeatures;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+public class NeoGlDeviceFeatures implements GpuDeviceFeatures {
```

Comment:
I would prefer if this was split into two separate classes:
- `MutableGpuDeviceFeatures` which is only used for enabled features, is backend-independent and takes another, pre-populated `GpuDeviceFeatures` instance for the available features
- `NeoGlDeviceFeatures` (i.e. a backend-specific impl of `GpuDeviceFeatures`) which collects and holds the available features and is passed to the former

Only the former would then have the setters for enabling features, would be exposed as the mutable type in the configuration event and later as the immutable type on the `GpuDevice` (can still add the locking if you're worried about people downcasting it).

Reply:
if it would just be adding a "Mutable" interface even if the impl doesn't change, sure, I don't have a super strong opinion either way.